src/CMakeLists.txt

```diff
@@ -1,3 +1,28 @@
+if(PARALLEL)
+  set(PTPLib_repository "https://github.com/usi-verification-and-security/partition-channel.git")
+  set(PTPLib_branch ${branch})
+  if(NOT PTPLib_branch)
+    set(PTPLib_branch master)
+  endif()
+
+  message("-- OpenSMT is fetching PTPLib from " ${PTPLib_repository})
+  message("-- PTPLib Branch : ${PTPLib_branch}")
+
+  include(FetchContent)
+  FetchContent_Declare(
+          PTPLib
+          GIT_REPOSITORY ${PTPLib_repository}
+          GIT_TAG ${PTPLib_branch}
+  )
+
+  FetchContent_MakeAvailable(PTPLib)
+
+  add_library(PTPLib::PTPLib ALIAS PTPLib)
+
+  message("-- PTPLib files have been written to: ${PTPLib_SOURCE_DIR}")
+  execute_process(COMMAND ${CMAKE_CURRENT_LIST_DIR}/remove_PTPLib_tests.sh ${PTPLib_SOURCE_DIR} RESULT_VARIABLE res)
```

Comment:
To remove unused files and directories of external libs, to make project lighter when building a docker image of OpenSMT or in case of SMTS, when SMTS would fetch openSMT and link against it, then removing SMTS's non-related parts, would make SMTS docker image lighter

Reply:
Isn't there some other way of achieving the same?  It feels dangerous to run shell scripts from a the build system.

Reply:
The docker image should contain only files needed for the run, no? So it should not contain redundant files anyway.
That should be taken care of when you are building the docker image from SMTS, not in OpenSMT's build configuration.

Reply:
I'll remove it and do it at docker file level, though the reason I did it like this was removing redundant files of external projects (OpenSMT) I thought would be automatic at cmake level (because of build directory name is unknown).